<launch>
    <arg name="map_file"      default="/home/ict/map/"/>
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="enable_sync"         value="true"/>
        <arg name="align_depth"         value="true"/>
        <arg name="enable_gyro"         value="true"/>
        <arg name="enable_accel"        value="true"/>
        <arg name="unite_imu_method"    value="copy"/>
    </include>
    <node pkg="tf" type="static_transform_publisher" name="rs_camera" args="1 0 0.35 0 0 0 1 map camera_link 50"/>
	<node pkg="depth2pointcloud_server" type="depth2pointcloud_server_node" name="depth2pointcloud_server" output="screen">
		<param name="depth_image" type="string" value="/camera/depth/image_rect_raw"/>	
	    <param name="camera_info" type="string" value="/camera/depth/camera_info" />
        <param name="downsample_scale" type="int" value="4" />
		<param name="leaf_size" type="double" value="0.02" />
		<param name="max_depth" type="double" value="3.0" />
		<param name="remove_outlier" type="bool" value="true" />
		<param name="remove_ground" type="bool" value="false" />
		<param name="fillhole_depth" type="bool" value="false" />
		<param name="add_airwall" type="bool" value="true" />
	</node>
	<node pkg="local_octomap_server" type="local_octomap_server_node" name="local_octomap_server" output="screen">
		<param name="pointcloud" type="string" value="/depth2pointcloud_server/pointcloud" />
		<param name="base_frame" type="string" value="map" />
        <param name="max_range" type="double" value="-1.0" />
        <param name="max_rayRange" type="double" value="3.3" />
        <param name="leaf_size" type="double" value="0.05" />
        <param name="publish_gridmap" type="bool" value="true" />
        <param name="loadmap_path" type="string" value="$(arg map_file)" />
        <param name="incremental_update" type="bool" value="true" />
        
	</node>
</launch>

